pyccel/parser/semantic.py

```diff
@@ -454,15 +450,16 @@ def _infere_type(self, expr, **settings):
             d_var['rank'          ] = expr.rank
             d_var['shape'         ] = expr.shape
             d_var['order'         ] = expr.order
-            d_var['is_stack_array'] = d.get('is_stack_array',False) and isinstance(expr.length, LiteralInteger)
-            d_var['allocatable'   ] = not d_var['is_stack_array']
-            d_var['is_pointer'    ] = False
             d_var['cls_base'      ] = TupleClass
+            if 'memory_handling' in d.keys() and d['memory_handling'] == 'stack' and isinstance(expr.length, LiteralInteger):
```

Comment:
Similarly to above this can be written more compactly using the default argument of the `get` function